### ISSUE TITLE
Licensing

### ISSUE THREAD

**Issue by arturalbov** [Belval/pdf2image#124]
Hello!

I want to use pdf2image in commercial project with closed source code. May be it's a stupid question but I'm a little bit confused that you're using `MIT` license while poppler is under `GPL` license. Could you please dispel my doubts?

Thanks!

**Comment by Belval** [comment]
Obligatory "I am not a lawyer", anything written below does not constitute legal advice, but there is still an easy answer to your question.

Poppler is distributed under a copyleft license, namely GPL 2.0. If in your commercial application you modify Poppler and/or do static/dynamic linking on Poppler you can be infringing on their license by not releasing your code source. You cannot distribute Poppler yourself publicly without disclosing your source code.

That being said, it is by design that `pdf2image` does not in anyway way or form, link directly to poppler. All it does is call an already existing CLI utility that comes pre-installed on most Linux installation. As such, my code is distributed under MIT and can be because it's the only thing that I am distributing (and none of the GPL bits). You can make any changes you want to `pdf2image`, even sell it without infringing on my or anyones' copyright.

TL;DR: It's okay because `pdf2image` is not built or distributed with Poppler none of the Python code is "contaminated" by the GPL license.

References: 
1. https://en.wikipedia.org/wiki/GNU_General_Public_License#Linking_and_derived_works
2. https://tldrlegal.com/license/gnu-general-public-license-v2
3. https://gitlab.freedesktop.org/poppler/poppler/blob/master/COPYING

**Comment by arturalbov** [comment]
Ok, I see. Thank you for explanation!

**Comment by aschmu** [comment]
Hello I'm reopening this since it's not clear to me if running pdf2image inside a docker container can still be considered as MIT compatible ??

**Comment by Belval** [comment]
Again not a lawyer, but this is a question that spans much wider than `pdf2image`: https://opensource.com/article/18/1/containers-gpl-and-copyleft.

TLDR: The container is not considered anything as it is not code, but GPL code can "live" with MIT code in a container without any issue.
